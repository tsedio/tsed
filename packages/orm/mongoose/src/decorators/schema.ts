import {decoratorTypeOf, StoreMerge, useDecorators} from "@tsed/core";
import {registerProvider} from "@tsed/di";
import {Property} from "@tsed/schema";
import {SchemaTypeOptions} from "mongoose";
import {MONGOOSE_SCHEMA} from "../constants";
import {MongooseSchemaOptions} from "../interfaces";
import {getSchema, getSchemaToken} from "../utils/createSchema";

/**
 * Define a class as a Mongoose Schema ready to be used to compose other schemes and models.
 *
 * ### Example
 *
 * ```typescript
 * @MongooseSchema()
 * export class EventSchema {
 *   @Property()
 *   field: string;
 * }
 * ```
 *
 * ### Options
 *
 * - `schemaOptions` (mongoose.SchemaOptions): Option to configure the schema behavior.
 *
 * @param {MongooseSchemaOptions | undefined} options
 * @returns {(target: any) => void}
 * @decorator
 * @mongoose
 * @property
 * @class
 */
export function Schema(options?: MongooseSchemaOptions): (target: any) => void;
export function Schema(definition: SchemaTypeOptions<any>): Function;
export function Schema(options: MongooseSchemaOptions | SchemaTypeOptions<any> = {}) {
  return (...parameters: any[]) => {
    switch (decoratorTypeOf(parameters)) {
      case "property":
        return useDecorators(Property(), StoreMerge(MONGOOSE_SCHEMA, options))(parameters[0], parameters[1], parameters[2]);

      case "class":
        const {token} = getSchemaToken(parameters[0], options);

        registerProvider({
          provide: token,
          deps: [],
          useFactory() {
            return getSchema(parameters[0], options as any);
          }
        });
        break;
    }
  };
}

/**
 * Define a class as a Mongoose Schema ready to be used to compose other schemes and models.
 *
 * ### Example
 *
 * ```typescript
 * @MongooseSchema()
 * export class EventSchema {
 *   @Property()
 *   field: string;
 * }
 * ```
 *
 * ### Options
 *
 * - `schemaOptions` (mongoose.SchemaOptions): Option to configure the schema behavior.
 *
 * @param {MongooseSchemaOptions | undefined} options
 * @returns {(target: any) => void}
 * @decorator
 * @mongoose
 * @property
 * @class
 */
export function MongooseSchema(options?: MongooseSchemaOptions): (target: any) => void;
export function MongooseSchema(definition: SchemaTypeOptions<any>): Function;
export function MongooseSchema(options: MongooseSchemaOptions | SchemaTypeOptions<any> = {}) {
  return Schema(options as any);
}
