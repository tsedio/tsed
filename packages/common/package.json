{
  "name": "@tsed/common",
  "version": "6.63.2",
  "description": "A TypeScript Framework on top of Express",
  "source": "./src/index.ts",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "private": false,
  "keywords": [
    "Express",
    "Koa",
    "TypeScript",
    "typescript",
    "Decorator",
    "decorators",
    "decorator",
    "express",
    "koa",
    "Controller",
    "Inject",
    "ioc",
    "di",
    "mvc",
    "swagger",
    "swagger ui",
    "ES2015",
    "ES6",
    "server",
    "rest",
    "api",
    "validation"
  ],
  "author": {
    "name": "Romain Lenzotti"
  },
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tsedio/tsed/issues"
  },
  "homepage": "http://tsed.io/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tsedio//tsed.git"
  },
  "scripts": {
    "build": "tsc --build tsconfig.compile.json"
  },
  "dependencies": {
    "@tsed/core": "6.63.2",
    "@tsed/perf": "6.63.2",
    "@tsed/di": "6.63.2",
    "@tsed/exceptions": "6.63.2",
    "@tsed/json-mapper": "6.63.2",
    "@tsed/logger": "^5.16.0",
    "@tsed/schema": "6.63.2",
    "@types/cache-manager": "^3.4.0",
    "@types/json-schema": "7.0.7",
    "cache-manager": "^3.4.1",
    "consolidate": "^0.16.0",
    "ejs": "^3.1.5",
    "json-schema": "^0.2.3",
    "on-finished": "2.3.0",
    "tslib": "2.2.0",
    "uuid": "8.3.2"
  },
  "devDependencies": {
    "@tsed/engines": "^1.1.4",
    "@types/consolidate": "0.14.0",
    "rxjs": "^6.5.2"
  },
  "peerDependencies": {
    "@tsed/engines": "^1.1.4"
  }
}