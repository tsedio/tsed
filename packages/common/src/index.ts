export * from "@tsed/di";
export * from "@tsed/platform-exceptions";
export * from "@tsed/platform-response-filter";
export * from "@tsed/platform-cache";
export * from "./config";
export * from "./mvc";
export * from "./platform";
export * from "./platform-test";

export {$log, Logger} from "@tsed/logger";
