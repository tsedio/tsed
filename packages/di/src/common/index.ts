/**
 * @file Automatically generated by @tsed/barrels.
 */
export * from "./constants/constants.js";
export * from "./decorators/autoInjectable.js";
export * from "./decorators/configuration.js";
export * from "./decorators/constant.js";
export * from "./decorators/controller.js";
export * from "./decorators/inject.js";
export * from "./decorators/injectable.js";
export * from "./decorators/intercept.js";
export * from "./decorators/interceptor.js";
export * from "./decorators/lazyInject.js";
export * from "./decorators/module.js";
export * from "./decorators/opts.js";
export * from "./decorators/overrideProvider.js";
export * from "./decorators/scope.js";
export * from "./decorators/service.js";
export * from "./decorators/useOpts.js";
export * from "./decorators/value.js";
export * from "./domain/Container.js";
export * from "./domain/ControllerProvider.js";
export * from "./domain/InjectablePropertyType.js";
export * from "./domain/LocalsContainer.js";
export * from "./domain/Provider.js";
export * from "./domain/ProviderScope.js";
export * from "./domain/ProviderType.js";
export * from "./errors/InjectionError.js";
export * from "./errors/InvalidPropertyTokenError.js";
export * from "./errors/UndefinedTokenError.js";
export * from "./interfaces/DIConfigurationOptions.js";
export * from "./interfaces/DILogger.js";
export * from "./interfaces/DILoggerOptions.js";
export * from "./interfaces/DIResolver.js";
export * from "./interfaces/ImportTokenProviderOpts.js";
export * from "./interfaces/InjectableProperties.js";
export * from "./interfaces/InterceptorContext.js";
export * from "./interfaces/InterceptorMethods.js";
export * from "./interfaces/InvokeOptions.js";
export * from "./interfaces/OnDestroy.js";
export * from "./interfaces/OnInit.js";
export * from "./interfaces/ProviderOpts.js";
export * from "./interfaces/RegistrySettings.js";
export * from "./interfaces/ResolvedInvokeOptions.js";
export * from "./interfaces/TokenProvider.js";
export * from "./interfaces/TokenRoute.js";
export * from "./registries/GlobalProviders.js";
export * from "./registries/ProviderRegistry.js";
export * from "./services/DIConfiguration.js";
export * from "./services/DILogger.js";
export * from "./services/InjectorService.js";
export * from "./utils/colors.js";
export * from "./utils/createContainer.js";
export * from "./utils/getConfiguration.js";
export * from "./utils/getConstructorDependencies.js";
export * from "./utils/resolveControllers.js";
