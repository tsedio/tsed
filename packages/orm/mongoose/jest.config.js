// For a detailed explanation regarding each configuration property, visit:
// https://jestjs.io/docs/en/configuration.html

module.exports = {
  ...require("@tsed/jest-config")(__dirname, "mongoose"),
  coverageThreshold: {
    global: {
      statements: 99.5,
      branches: 95.48,
      functions: 100,
      lines: 99.47
    }
  }
};
