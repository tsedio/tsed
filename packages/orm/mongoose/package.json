{
  "name": "@tsed/mongoose",
  "description": "Mongoose package for Ts.ED framework",
  "type": "commonjs",
  "version": "7.83.4",
  "keywords": [
    "tsed",
    "mongoose",
    "typescript",
    "decorators",
    "orm"
  ],
  "source": "./src/index.ts",
  "main": "./lib/cjs/index.js",
  "module": "./lib/esm/index.js",
  "typings": "./lib/types/index.d.ts",
  "exports": {
    ".": {
      "types": "./lib/types/index.d.ts",
      "import": "./lib/esm/index.js",
      "require": "./lib/cjs/index.js",
      "default": "./lib/cjs/index.js"
    }
  },
  "scripts": {
    "build": "yarn barrels && yarn build:ts",
    "barrels": "barrels",
    "test": "vitest run",
    "build:ts": "tsc --build tsconfig.json",
    "test:ci": "vitest run --coverage.thresholds.autoUpdate=true"
  },
  "dependencies": {
    "change-case": "4.1.2",
    "tslib": "2.6.1"
  },
  "devDependencies": {
    "@tsed/ajv": "workspace:*",
    "@tsed/barrels": "workspace:*",
    "@tsed/core": "workspace:*",
    "@tsed/di": "workspace:*",
    "@tsed/json-mapper": "workspace:*",
    "@tsed/logger": ">=6.7.5",
    "@tsed/schema": "workspace:*",
    "@tsed/testcontainers-mongo": "workspace:*",
    "@tsed/typescript": "workspace:*",
    "eslint": "^8.57.0",
    "mongoose": "8.7.1",
    "typescript": "4.9.5",
    "vitest": "2.0.4"
  },
  "peerDependencies": {
    "@tsed/ajv": "7.83.4",
    "@tsed/core": "7.83.4",
    "@tsed/di": "7.83.4",
    "@tsed/json-mapper": "7.83.4",
    "@tsed/logger": ">=6.7.5",
    "@tsed/schema": "7.83.4",
    "mongoose": ">=6.0.0"
  }
}
