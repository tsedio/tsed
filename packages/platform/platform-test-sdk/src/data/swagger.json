{
  "openapi": "3.0.1",
  "info": {
    "version": "1.0.0",
    "description": "Swagger description",
    "title": "Swagger Title",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "email": "apiteam@swagger.io"
    },
    "license": {
      "name": "Apache 2.0",
      "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
    }
  },
  "components": {
    "securitySchemes": {
      "global_auth": {
        "type": "oauth2",
        "flows": {
          "implicit": {
            "authorizationUrl": "http://petstore.swagger.io/oauth/dialog",
            "scopes": {
              "admin": "Admin access"
            }
          }
        }
      }
    }
  },
  "paths": {}
}
