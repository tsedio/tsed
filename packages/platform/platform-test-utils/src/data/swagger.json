{
  "swagger": "2.0",
  "info": {
    "version": "1.0.0",
    "description": "Swagger description",
    "title": "Swagger Title",
    "termsOfService": "http://swagger.io/terms/",
    "contact": {
      "email": "apiteam@swagger.io"
    },
    "license": {
      "name": "Apache 2.0",
      "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
    }
  },
  "securityDefinitions": {
    "global_auth": {
      "type": "oauth2",
      "authorizationUrl": "http://petstore.swagger.io/oauth/dialog",
      "flow": "implicit",
      "scopes": {
        "admin": "Admin access"
      }
    }
  },
  "paths": {}
}
