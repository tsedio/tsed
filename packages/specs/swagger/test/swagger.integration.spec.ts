import {BodyParams, Controller, Get, PathParams, PlatformTest, Post} from "@tsed/common";
import {ObjectID} from "@tsed/mongoose";
import {MergeParams, PlatformExpress} from "@tsed/platform-express";
import {Consumes, Description, Returns} from "@tsed/schema";
import {Docs, Hidden} from "@tsed/swagger";
import {expect} from "chai";
import SuperTest from "supertest";
import {Calendar} from "./app/models/Calendar";
import {Server} from "./app/Server";

@Controller("/admin")
@Hidden()
class AdminCtrl {
  @Get("/")
  get() {}
}

@Controller("/events")
@MergeParams(true)
class EventCtrl {
  @Get("/")
  @Description("Events")
  get() {}
}

@Controller("/admin")
@Docs("admin")
class BackAdminCtrl {
  @Get("/")
  @Description("Admins")
  get() {}
}

@Controller({
  path: "/calendars",
  children: [AdminCtrl, EventCtrl]
})
class CalendarsController {
  @Get("/:id")
  @Returns(200, Calendar)
  async get(@PathParams("id") @ObjectID() id: string): Promise<Calendar> {
    return new Calendar({id, name: "test"});
  }

  @Get("/")
  @(Returns(200, Array).Of(Calendar))
  async getAll(): Promise<Calendar[]> {
    return [new Calendar({id: 1, name: "name"}), new Calendar({id: 2, name: "name"})];
  }

  @Post("/csv")
  @Consumes("text/plain")
  @(Returns(200, String).ContentType("text/plain"))
  async csv(@BodyParams() csvLines: string): Promise<string> {
    return "";
  }

  @Get("/hidden")
  @Hidden()
  @Description("Admins")
  getHidden() {}
}

describe("Swagger integration", () => {
  describe("OpenSpec2", () => {
    let request: SuperTest.SuperTest<SuperTest.Test>;
    beforeEach(
      PlatformTest.bootstrap(Server, {
        platform: PlatformExpress,
        mount: {
          "/rest": [CalendarsController]
        }
      })
    );
    beforeEach(() => {
      request = SuperTest(PlatformTest.callback());
    });
    afterEach(PlatformTest.reset);

    it("should swagger spec", async () => {
      const response = await request.get("/v2/doc/swagger.json").expect(200);
      const result = await request.get("/rest/calendars").expect(200);

      expect(result.body).to.deep.eq([
        {
          id: 1,
          name: "name"
        },
        {
          id: 2,
          name: "name"
        }
      ]);
      expect(response.body).to.deep.eq({
        consumes: ["application/json"],
        definitions: {
          Calendar: {
            properties: {
              id: {
                type: "string"
              },
              name: {
                minLength: 1,
                type: "string"
              }
            },
            required: ["name"],
            type: "object"
          }
        },
        info: {
          title: "Swagger title",
          version: "1.2.0"
        },
        paths: {
          "/rest/calendars": {
            get: {
              operationId: "calendarsControllerGetAll",
              parameters: [],
              produces: ["application/json"],
              responses: {
                "200": {
                  description: "Success",
                  schema: {
                    items: {
                      $ref: "#/definitions/Calendar"
                    },
                    type: "array"
                  }
                }
              },
              tags: ["CalendarsController"]
            }
          },
          "/rest/calendars/{id}": {
            get: {
              operationId: "calendarsControllerGet",
              parameters: [
                {
                  description: "Mongoose ObjectId",
                  in: "path",
                  name: "id",
                  pattern: "^[0-9a-fA-F]{24}$",
                  required: true,
                  type: "string"
                }
              ],
              produces: ["application/json"],
              responses: {
                "200": {
                  description: "Success",
                  schema: {
                    $ref: "#/definitions/Calendar"
                  }
                }
              },
              tags: ["CalendarsController"]
            }
          },
          "/rest/calendars/csv": {
            post: {
              operationId: "calendarsControllerCsv",
              parameters: [
                {
                  in: "body",
                  name: "body",
                  required: false,
                  type: "string"
                }
              ],
              produces: ["text/plain"],
              consumes: ["text/plain"],
              responses: {
                "200": {
                  description: "Success",
                  schema: {
                    type: "string"
                  }
                }
              },
              tags: ["CalendarsController"]
            }
          },
          "/rest/calendars/events": {
            get: {
              description: "Events",
              operationId: "eventCtrlGet",
              parameters: [],
              responses: {
                "200": {
                  description: "Success"
                }
              },
              tags: ["EventCtrl"]
            }
          }
        },
        produces: ["application/json"],
        swagger: "2.0",
        tags: [
          {
            name: "EventCtrl"
          },
          {
            name: "CalendarsController"
          }
        ]
      });
    });
  });
  describe("OpenSpec3", () => {
    let request: SuperTest.SuperTest<SuperTest.Test>;
    beforeEach(
      PlatformTest.bootstrap(Server, {
        platform: PlatformExpress,
        mount: {
          "/rest": [CalendarsController]
        }
      })
    );
    beforeEach(() => {
      request = SuperTest(PlatformTest.callback());
    });
    afterEach(PlatformTest.reset);

    it("should swagger spec", async () => {
      const response = await request.get("/v3/doc/swagger.json").expect(200);
      const result = await request.get("/rest/calendars").expect(200);

      expect(result.body).to.deep.eq([
        {
          id: 1,
          name: "name"
        },
        {
          id: 2,
          name: "name"
        }
      ]);

      expect(response.body).to.deep.eq({
        info: {version: "1.0.0", title: "Api documentation"},
        openapi: "3.0.1",
        paths: {
          "/rest/calendars/events": {
            get: {
              operationId: "eventCtrlGet",
              parameters: [],
              responses: {"200": {description: "Success"}},
              description: "Events",
              tags: ["EventCtrl"]
            }
          },
          "/rest/calendars/{id}": {
            get: {
              operationId: "calendarsControllerGet",
              parameters: [
                {
                  description: "Mongoose ObjectId",
                  in: "path",
                  name: "id",
                  required: true,
                  schema: {
                    example: "5ce7ad3028890bd71749d477",
                    pattern: "^[0-9a-fA-F]{24}$",
                    type: "string"
                  }
                }
              ],
              responses: {
                "200": {
                  content: {"application/json": {schema: {$ref: "#/components/schemas/Calendar"}}},
                  description: "Success"
                }
              },
              tags: ["CalendarsController"]
            }
          },
          "/rest/calendars": {
            get: {
              operationId: "calendarsControllerGetAll",
              parameters: [],
              responses: {
                "200": {
                  content: {
                    "application/json": {
                      schema: {
                        type: "array",
                        items: {$ref: "#/components/schemas/Calendar"}
                      }
                    }
                  },
                  description: "Success"
                }
              },
              tags: ["CalendarsController"]
            }
          },
          "/rest/calendars/csv": {
            post: {
              operationId: "calendarsControllerCsv",
              parameters: [],
              requestBody: {
                content: {
                  "text/plain": {
                    schema: {
                      type: "string"
                    }
                  }
                },
                required: false
              },
              responses: {
                "200": {
                  content: {
                    "text/plain": {
                      schema: {
                        type: "string"
                      }
                    }
                  },
                  description: "Success"
                }
              },
              tags: ["CalendarsController"]
            }
          }
        },
        tags: [{name: "EventCtrl"}, {name: "CalendarsController"}],
        components: {
          schemas: {
            Calendar: {
              type: "object",
              properties: {id: {type: "string"}, name: {type: "string", minLength: 1}},
              required: ["name"]
            }
          }
        }
      });
    });
  });
});
