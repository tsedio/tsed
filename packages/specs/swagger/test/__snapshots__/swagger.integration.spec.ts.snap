// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Swagger integration > OpenSpec2 > should swagger spec 1`] = `
{
  "consumes": [
    "application/json",
  ],
  "definitions": {
    "Calendar": {
      "properties": {
        "id": {
          "type": "string",
        },
        "name": {
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
  },
  "info": {
    "title": "Swagger title",
    "version": "1.2.0",
  },
  "paths": {
    "/rest/calendars": {
      "get": {
        "operationId": "calendarsControllerGetAll",
        "parameters": [],
        "produces": [
          "application/json",
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "items": {
                "$ref": "#/definitions/Calendar",
              },
              "type": "array",
            },
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/csv": {
      "post": {
        "consumes": [
          "text/plain",
        ],
        "operationId": "calendarsControllerCsv",
        "parameters": [
          {
            "in": "body",
            "name": "body",
            "required": false,
            "schema": {
              "type": "string",
            },
          },
        ],
        "produces": [
          "text/plain",
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "type": "string",
            },
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/events": {
      "get": {
        "description": "Events",
        "operationId": "eventCtrlGet",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "EventCtrl",
        ],
      },
    },
    "/rest/calendars/{id}": {
      "get": {
        "operationId": "calendarsControllerGet",
        "parameters": [
          {
            "description": "An ObjectID",
            "in": "path",
            "name": "id",
            "pattern": "^[0-9a-fA-F]{24}$",
            "required": true,
            "type": "string",
          },
        ],
        "produces": [
          "application/json",
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "$ref": "#/definitions/Calendar",
            },
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
  },
  "produces": [
    "application/json",
  ],
  "swagger": "2.0",
  "tags": [
    {
      "name": "CalendarsController",
    },
    {
      "name": "EventCtrl",
    },
  ],
}
`;

exports[`Swagger integration > OpenSpec3 > should swagger spec 1`] = `
{
  "components": {
    "schemas": {
      "Calendar": {
        "properties": {
          "id": {
            "type": "string",
          },
          "name": {
            "minLength": 1,
            "type": "string",
          },
        },
        "required": [
          "name",
        ],
        "type": "object",
      },
    },
  },
  "info": {
    "title": "Api documentation",
    "version": "1.0.0",
  },
  "openapi": "3.0.1",
  "paths": {
    "/rest/calendars": {
      "get": {
        "operationId": "calendarsControllerGetAll",
        "parameters": [],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "items": {
                    "$ref": "#/components/schemas/Calendar",
                  },
                  "type": "array",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/csv": {
      "post": {
        "operationId": "calendarsControllerCsv",
        "parameters": [],
        "requestBody": {
          "content": {
            "text/plain": {
              "schema": {
                "type": "string",
              },
            },
          },
          "required": false,
        },
        "responses": {
          "200": {
            "content": {
              "text/plain": {
                "schema": {
                  "type": "string",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/events": {
      "get": {
        "description": "Events",
        "operationId": "eventCtrlGet",
        "parameters": [],
        "responses": {
          "200": {
            "description": "Success",
          },
        },
        "tags": [
          "EventCtrl",
        ],
      },
    },
    "/rest/calendars/{id}": {
      "get": {
        "operationId": "calendarsControllerGet",
        "parameters": [
          {
            "description": "An ObjectID",
            "in": "path",
            "name": "id",
            "required": true,
            "schema": {
              "example": "5ce7ad3028890bd71749d477",
              "pattern": "^[0-9a-fA-F]{24}$",
              "type": "string",
            },
          },
        ],
        "responses": {
          "200": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Calendar",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": [
          "CalendarsController",
        ],
      },
    },
  },
  "tags": [
    {
      "name": "CalendarsController",
    },
    {
      "name": "EventCtrl",
    },
  ],
}
`;

exports[`Swagger integration OpenSpec2 should swagger spec 1`] = `
Object {
  "consumes": Array [
    "application/json",
  ],
  "definitions": Object {
    "Calendar": Object {
      "properties": Object {
        "id": Object {
          "type": "string",
        },
        "name": Object {
          "minLength": 1,
          "type": "string",
        },
      },
      "required": Array [
        "name",
      ],
      "type": "object",
    },
  },
  "info": Object {
    "title": "Swagger title",
    "version": "1.2.0",
  },
  "paths": Object {
    "/rest/calendars": Object {
      "get": Object {
        "operationId": "calendarsControllerGetAll",
        "parameters": Array [],
        "produces": Array [
          "application/json",
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
            "schema": Object {
              "items": Object {
                "$ref": "#/definitions/Calendar",
              },
              "type": "array",
            },
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/csv": Object {
      "post": Object {
        "consumes": Array [
          "text/plain",
        ],
        "operationId": "calendarsControllerCsv",
        "parameters": Array [
          Object {
            "in": "body",
            "name": "body",
            "required": false,
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "produces": Array [
          "text/plain",
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
            "schema": Object {
              "type": "string",
            },
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/events": Object {
      "get": Object {
        "description": "Events",
        "operationId": "eventCtrlGet",
        "parameters": Array [],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "EventCtrl",
        ],
      },
    },
    "/rest/calendars/{id}": Object {
      "get": Object {
        "operationId": "calendarsControllerGet",
        "parameters": Array [
          Object {
            "description": "An ObjectID",
            "in": "path",
            "name": "id",
            "pattern": "^[0-9a-fA-F]{24}$",
            "required": true,
            "type": "string",
          },
        ],
        "produces": Array [
          "application/json",
        ],
        "responses": Object {
          "200": Object {
            "description": "Success",
            "schema": Object {
              "$ref": "#/definitions/Calendar",
            },
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
  },
  "produces": Array [
    "application/json",
  ],
  "swagger": "2.0",
  "tags": Array [
    Object {
      "name": "CalendarsController",
    },
    Object {
      "name": "EventCtrl",
    },
  ],
}
`;

exports[`Swagger integration OpenSpec3 should swagger spec 1`] = `
Object {
  "components": Object {
    "schemas": Object {
      "Calendar": Object {
        "properties": Object {
          "id": Object {
            "type": "string",
          },
          "name": Object {
            "minLength": 1,
            "type": "string",
          },
        },
        "required": Array [
          "name",
        ],
        "type": "object",
      },
    },
  },
  "info": Object {
    "title": "Api documentation",
    "version": "1.0.0",
  },
  "openapi": "3.0.1",
  "paths": Object {
    "/rest/calendars": Object {
      "get": Object {
        "operationId": "calendarsControllerGetAll",
        "parameters": Array [],
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "items": Object {
                    "$ref": "#/components/schemas/Calendar",
                  },
                  "type": "array",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/csv": Object {
      "post": Object {
        "operationId": "calendarsControllerCsv",
        "parameters": Array [],
        "requestBody": Object {
          "content": Object {
            "text/plain": Object {
              "schema": Object {
                "type": "string",
              },
            },
          },
          "required": false,
        },
        "responses": Object {
          "200": Object {
            "content": Object {
              "text/plain": Object {
                "schema": Object {
                  "type": "string",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
    "/rest/calendars/events": Object {
      "get": Object {
        "description": "Events",
        "operationId": "eventCtrlGet",
        "parameters": Array [],
        "responses": Object {
          "200": Object {
            "description": "Success",
          },
        },
        "tags": Array [
          "EventCtrl",
        ],
      },
    },
    "/rest/calendars/{id}": Object {
      "get": Object {
        "operationId": "calendarsControllerGet",
        "parameters": Array [
          Object {
            "description": "An ObjectID",
            "in": "path",
            "name": "id",
            "required": true,
            "schema": Object {
              "example": "5ce7ad3028890bd71749d477",
              "pattern": "^[0-9a-fA-F]{24}$",
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "200": Object {
            "content": Object {
              "application/json": Object {
                "schema": Object {
                  "$ref": "#/components/schemas/Calendar",
                },
              },
            },
            "description": "Success",
          },
        },
        "tags": Array [
          "CalendarsController",
        ],
      },
    },
  },
  "tags": Array [
    Object {
      "name": "CalendarsController",
    },
    Object {
      "name": "EventCtrl",
    },
  ],
}
`;
